backend/common/src/most_seen.rs

```diff
@@ -0,0 +1,247 @@
+// Source code for the Substrate Telemetry Server.
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::hash::Hash;
+
+/// Add items to this, and it will keep track of what the item
+/// seen the most is.
+#[derive(Debug)]
+pub struct MostSeen<T> {
+    current_best: T,
+    current_count: usize,
+    others: HashMap<T, usize>,
+}
+
+impl<T: Default> Default for MostSeen<T> {
+    fn default() -> Self {
```

Comment:
hrm, can you explain the difference between `new()` and `default()` such that why is `current_count: 0` in default and `current_count: 1` in new?

Reply:
I should definitely add a comment about this because it confused me too at first!

`new` is given an actual item (like the name of a chain), so it logs that we've seen that item once (because we were just handed it).

`default` hasn't been given any items yet, so the "current best" that you'd get back is the default value of the type, and it's never actually been seen by anybody. So the first time you add any item, it becomes the "most seen item" with a count of 1.